### DISCUSSION TITLE
Unable to find package Tmds.DBus.SourceGenerator

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#10871] [Q&A] [answered]
After upgrading to latest nightly (11.0.999-cibuild0032690-beta) from 11.0-preview6, and then fixing the breaking changes, I get the following build error:

```
Error	NU1101	Unable to find package Tmds.DBus.SourceGenerator. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, Package source
```

Anyone have ideas how to fix this? Am I missing something simple?

 1. I thought this is only needed for Linux and I'm on a Windows PC at the moment. It seems this package shouldn't be pulled in.
 2. This isn't a package I see on Nuget either, do I need to add a package source? (I really shouldn't have to for vanilla build)
 3. Did something change with how project files should be configured?

Here is repo: https://github.com/affederaffe/Tmds.DBus.SourceGenerator

cc @affederaffe

**Comment by kekekeks** [comment, marked as answer]
1) Package [definitely exists on nuget.org](https://www.nuget.org/packages/Tmds.DBus.SourceGenerator), try clearing your nuget caches
2) this `PackageReference` should have `PrivateAssets="all"`  https://github.com/AvaloniaUI/Avalonia/blob/6dfc2cadf44e2f4b2e07a19085880827dcb48ec3/src/Avalonia.FreeDesktop/Avalonia.FreeDesktop.csproj#L16

**Reply by robloo** [reply]
> Package definitely exists on nuget.org, try clearing your nuget caches

I'm not sure it's published correctly. I tried a few different things and while it is on Nuget.org as you pointed out, it isn't available in Visual Studio. It also doesn't have any target framework versions assigned to it.

> this PackageReference should have PrivateAssets="all"

Thanks, going to do a PR for that soon.

**Reply by robloo** [reply]
The `PrivateAssets="all"` PR seems to have fixed this issue.
